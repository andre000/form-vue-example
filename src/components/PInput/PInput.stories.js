import '@/assets/scss/storybook.scss'
import { withKnobs, text, boolean } from '@storybook/addon-knobs'
import PInput from './PInput.vue'

export default {
  component: PInput,
  title: 'PInput',
  decorator: [withKnobs]
}

export const withInitialState = () => ({
  components: { PInput },
  template: '<p-input />'
})

export const withErrorMessage = () => ({
  components: { PInput },
  props: {
    text: {
      default: text('Error', 'This is an error')
    }
  },
  template: '<p-input :errorMessage="text" />'
})

export const withMoneyMask = () => ({
  components: { PInput },
  props: {
    money: {
      default: boolean('Money mask', true)
    }
  },
  template: '<p-input :is-money="money" />'
})
