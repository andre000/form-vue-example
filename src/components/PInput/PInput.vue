<template>
  <div class="field" :class="errorMessage && 'field--error'">
    <input v-on="$listeners" v-bind="$attrs" />
    <div v-if="errorMessage" class="field__message field__message--error">
      {{ errorMessage }}
    </div>
  </div>
</template>

<script>
export default {
  inheritAttrs: false,
  props: {
    value: {
      type: [String, Number]
    },

    errorMessage: {
      type: String,
      default: null
    },

    mask: {
      type: String,
      default: null
    }
  },

  computed: {
    listeners() {
      return {
        ...this.$listeners,
        input: this.updateValue
      }
    }
  },

  methods: {
    updateValue(e) {
      this.$emit('input', e.target.value)
    }
  }
}
</script>

<style></style>
