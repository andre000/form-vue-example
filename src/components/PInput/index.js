export { default as PInput } from './PInput.vue'
