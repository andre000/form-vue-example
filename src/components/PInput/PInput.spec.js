import { shallowMount, createLocalVue } from '@vue/test-utils'

import PInput from './PInput'
import { Money } from 'v-money'

const localVue = createLocalVue()

describe('PInput component', () => {
  let wrapper
  beforeAll(() => {
    wrapper = shallowMount(PInput, {
      localVue
    })
  })

  it('should match the snapshot', () =>
    expect(wrapper.html()).toMatchSnapshot())

  it('should have an input element', () => {
    expect(wrapper.find('input').exists()).toBe(true)
  })

  it('should display an error when the "errorMessage" prop has value', async () => {
    await wrapper.setProps({ errorMessage: 'This is an error' })
    expect(wrapper.find('.field__message--error').exists()).toBe(true)
  })

  it('should add "field--error" class to the component when an error is displayed', async () => {
    await wrapper.setProps({ errorMessage: 'This is an error' })
    expect(wrapper.classes('field--error')).toBe(true)
  })

  it('should render "v-money" component when "is-money" is true', async () => {
    await wrapper.setProps({ isMoney: true })
    expect(wrapper.findComponent(Money).exists()).toBe(true)
  })
})
