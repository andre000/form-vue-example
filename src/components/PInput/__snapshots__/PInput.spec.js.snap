// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PInput component should match the snapshot 1`] = `
<div class="field"><input>
  <!---->
</div>
`;
